### ISSUE TITLE
Add .NET support

### ISSUE THREAD

**Issue by ThePhantomMask** [SagaPDev/Godot-SDL-Gyro#1]
Is there also the possibility of adding support for .NET/C# programming in addition to GDScript?

**Comment by SagaPDev** [comment]
I'm not familiar using c# with the mono build of Godot,  do you get any errors when trying to make "SDLGyro.new()" from a C# script?

From what i know functions added by GDextention should behave as regular built-in Godot functions.

**Comment by ThePhantomMask** [comment]
It just doesn't work, I think it must be made on purpose the combinability.

**Comment by SagaPDev** [comment]
Yes apparently it is true, but there is not much official documentation around it, just discussions and proposed solutions, i will take a look and post here if there is any progress, if you find anything that could be helpful feel free to link it.

**Comment by ThePhantomMask** [comment]
And then on the GDScript side: I get this error: modules/gdscript/gdscript_compiler.cpp:121 - GDScript bug: Native class 'SDLGyro' not found.

**Comment by SagaPDev** [comment]
[sdl_mono_test.zip](https://github.com/SagaPDev/Godot-SDL-Gyro/files/14894040/sdl_mono_test.zip)
make sure you place the addon folder in the root of your project (here is an example i made with the mono build of godot but still using gdscript)

**Comment by ThePhantomMask** [comment]
On my project it gives me the problem, on your project it does not give the same problems

**Comment by ThePhantomMask** [comment]
getCalibratedGyro() and getProcessedAcceleration() these two methods do not seem to work.

**Comment by SagaPDev** [comment]
Remember that is necessary to call gamepadPolling() for the other functions to work (like this):

```
...
func _process(_delta):
  Gyro.gamepadPolling()
  print(Gyro.getCalibratedGyro())
...
```

**Comment by ThePhantomMask** [comment]
A suggestion:
Integrating vectors (2/3/4) would be convenient, in standard C++ they do not exist but there are external libraries that make integration easier for you: GLM or Eigen

**Comment by SagaPDev** [comment]
could you elaborate on that?

**Comment by ThePhantomMask** [comment]
Instead of the typed array you could use Vector2/3/4 of Godot

**Comment by SagaPDev** [comment]
I tried it at first but, but gamepadMotionHelperes expected floats so i ended up using the Typed Array instead, i considered converting the array to a vector and returning it but didn't want to put more chain in the process. I will revisit this idea

**Comment by ThePhantomMask** [comment]
OK...got it.
Have you tried putting a vector that is initialised with the array data and then return the vector

**Comment by SagaPDev** [comment]
This is the exact thing i want to try, (i'm currently doing some freelance job, i will add this to the list on changes i have planed when i come back)

**Comment by ThePhantomMask** [comment]
But where should I put SDL if I want to compile it as a GDExstension file?

**Comment by SagaPDev** [comment]
if you're in windows, inside this project folder you will have to make the libs folder and then the mingw_dev_lib folder inside of it, place the files and folders extracted from SDL2-X.XX.X\x86_64-w64-mingw32 there)
i hope i was clear, maybe i should change this

**Comment by ThePhantomMask** [comment]
Meanwhile on my fork I am trying to add Vector3 and 4

**Comment by ThePhantomMask** [comment]
> if you're in windows, inside this project folder you will have to make the libs folder and then the mingw_dev_lib folder inside of it, place the files and folders extracted from SDL2-X.XX.X\x86_64-w64-mingw32 there) i hope i was clear, maybe i should change this

But within src folder?

**Comment by SagaPDev** [comment]
no no, in the root of your project, take a look:
![1714955981_grim](https://github.com/SagaPDev/Godot-SDL-Gyro/assets/43705757/eb7151ab-a007-4a11-bc77-c52ddedf87ba)

**Comment by ThePhantomMask** [comment]
> no no, in the root of your project, take a look:
> ![1714955981_grim](https://github.com/SagaPDev/Godot-SDL-Gyro/assets/43705757/eb7151ab-a007-4a11-bc77-c52ddedf87ba)
> 

 Then I have to put the folder with all the SDL2 functions on src, right?
But how to solve a fatal error where it is impossible to open SDL2.lib how do I do it?
